services/property-svc/migrations/000002_property_values.up.sql

```diff
@@ -0,0 +1,52 @@
+CREATE OR REPLACE FUNCTION validate_select_option(property_id uuid, select_value uuid)
+RETURNS BOOLEAN AS $$
+BEGIN
+	IF select_value IS NOT NULL THEN
+		RETURN EXISTS (
+			SELECT 1
+	   		FROM properties
+	   			RIGHT JOIN select_datas ON properties.select_data_id = select_datas.id
+				RIGHT JOIN select_options ON select_datas.id = select_options.select_data_id
+			WHERE properties.id = property_id AND select_options.id = select_value
+		);
+	ELSE
+		RETURN TRUE;
+	END IF;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE TABLE IF NOT EXISTS property_values (
+	id uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+	property_id uuid NOT NULL,
+	subject_id uuid NOT NULL,
```

Comment:
we frequently do lookups by `(property_id, subjectid)` so we should add an [Index](https://www.postgresql.org/docs/current/sql-createindex.html)